### PR TITLE
[SYCL] Don't create events in piEnqueueKernelLaunch if possible

### PR DESCRIPTION
**Pull request by alexanderfle**
Signed-off-by: Alexander Flegontov <alexander.flegontov@intel.com>

### CODE REVIEW

File: sycl/source/detail/scheduler/commands.cpp [hunk 1]
```diff
@@ -1773,7 +1773,8 @@ pi_result ExecCGCommand::SetKernelParamsAndLaunch(
   pi_result Error = Plugin.call_nocheck<PiApiKind::piEnqueueKernelLaunch>(
       MQueue->getHandleRef(), Kernel, NDRDesc.Dims, &NDRDesc.GlobalOffset[0],
       &NDRDesc.GlobalSize[0], LocalSize, RawEvents.size(),
-      RawEvents.empty() ? nullptr : &RawEvents[0], &Event);
+      RawEvents.empty() ? nullptr : &RawEvents[0],
+      MQueue->is_event_required() ? &Event : nullptr);
```
Comment (romanovvlad) [inline sycl/source/detail/scheduler/commands.cpp:1777]: Could you please clarify why this change is legal? What if a user stores a `sycl::event` object returned by `queue::submit` and call `event::wait` or pass it to the `handler::depends_on`?

Comment (alexanderfle) [reply sycl/source/detail/scheduler/commands.cpp:1777]: This is just a test run so that it highlights all the problems and it does not belong to the final solution yet.
So any suggestions for improving this to make it work are welcome!
